feat(user): implement address management system

ykxin opened the pull request:
## 描述
本 PR 实现了用户个人中心的全栈地址管理功能，允许用户添加、编辑、删除及管理常用收货地址。

## 相关问题

无相关问题

## 变更类型

- [ ] Bug 修复（不会破坏现有功能的非破坏性变更）
- [x] 新功能（添加功能的非破坏性变更）
- [ ] 破坏性变更（会导致现有功能无法正常工作的修复或功能）
- [ ] 文档更新
- [ ] 性能改进
- [ ] 代码重构
- [ ] CI/CD 改进

## 测试

- [x] 我已经在本地测试了这些更改
- [x] 我已经添加了证明我的修复有效或我的功能工作的测试
- [x] 新的和现有的单元测试在我的更改下都能通过
- [x] 我已经检查了代码能够无错误地构建

## 截图（如果适用）

<img width="695" height="611" alt="image" src="https://github.com/user-attachments/assets/404750d9-71bd-4e5e-b171-98531a465c51" />


## 检查清单

- [x] 我的代码遵循此项目的代码风格
- [x] 我已经对自己的代码进行了自我审查
- [x] 我已经对代码进行了注释，特别是在难以理解的地方
- [x] 我已经对文档进行了相应的更改
- [x] 我的更改没有产生新的警告
- [x] 我已经添加了证明我的修复有效或我的功能工作的测试
- [x] 新的和现有的单元测试在我的更改下都能通过

## 代码审查检查清单（供审查者使用）

- [ ] 代码可读且有良好的文档
- [ ] 更改经过了适当的测试
- [ ] 没有明显的性能问题
- [ ] 已解决安全考虑
- [ ] 破坏性变更已得到适当记录

## 附加说明

在这里添加任何附加说明或背景信息。

ykxin committed:
feat(user): implement address management system

- Implement address CRUD operations in repository and server actions
- Add AddressForm and AddressList UI components
- Support setting and cancelling default addresses
- Add Zod schema validation and type definitions
- Fix TypeScript errors in form control and submission handlers
- Add integration tests for address repository

dzx0902 commented on .env.test:
```diff
@@ -5,7 +5,7 @@
 # Format: postgresql://[username]:[password]@[host]:[port]/[database_name]
 # Example: postgresql://user:password@localhost:5432/nomad_db
-DATABASE_URL=postgresql://localhost:5432/nomad_test
+DATABASE_URL=postgresql://postgres:123456@localhost:5432/nomad_test
```
这个最好不要提交上来，有安全性问题

dzx0902 commented on scripts/create-test-db.ts:
```diff
@@ -0,0 +1,33 @@
+import { Client } from "pg";
+
+const connectionString = "postgresql://postgres:123456@localhost:5432/nomad";
```
这个和上面是一样的问题，这里最好改成对环境文件的引用来避免硬编码，写出测试本地数据库

dzx0902 replied:
还有这个文件是什么作用，必须的么？

dzx0902 commented on src/domains/user/address.repository.ts:
```diff
@@ -0,0 +1,141 @@
+      .update(addresses)
+      .set({ isDefault: false })
+      .where(eq(addresses.userId, data.userId));
+  }
```
这个问题比较重要，会有较大的安全隐患，请解决

dzx0902 commented on src/domains/user/address.repository.ts:
```diff
@@ -0,0 +1,141 @@
+      .update(addresses)
+      .set({ isDefault: false })
+      .where(and(eq(addresses.userId, userId), ne(addresses.id, addressId)));
+  }
```
同上

ykxin committed:
fix(user): enhance address management security and performance

- Fix race condition in create/update address using DB transactions
- Optimize first-address check using COUNT query instead of fetching all rows
- Improve Zod validation error handling to return all issues
- Remove unsafe type assertions in AddressForm and Server Actions
- Enhance error logging and user feedback in frontend
- Secure test environment by removing hardcoded credentials

ykxin committed:
fix(security): revert .env.test  and .gitignore to old version

ukeSJTU commented on app/(frontend)/(without-sidebar)/home/info/page.tsx:
```diff
@@ -2,8 +2,9 @@ import Link from "next/link";
-import { UserInfoForm } from "@/components/user";
+import { AddressList, UserInfoForm } from "@/components/user";
 import { getUserInfo } from "@/domains/user";
+import { getUserAddresses } from "@/domains/user/address.repository";
```
这里应该从 `@/domains/user` 里面导入，保持类似的代码风格。可能需要调整 `src/domains/user/index.ts` 文件

ukeSJTU reviewed (changes requested):
整体功能实现很完整，代码逻辑清晰，只有一个非常小的代码风格相关的需要修改

ykxin committed:
refactor(user): standardize domain imports and file structure

- Export address repository methods from domain index
- Update page and action imports to use domain barrel file
- Remove direct repository imports in frontend components
- Align project structure with architectural standards

ukeSJTU reviewed (approved):
LGTM
